src/main/scala/nl/knaw/dans/easy/solr/SolrDocumentGenerator.scala

```diff
@@ -15,29 +15,119 @@
  */
 package nl.knaw.dans.easy.solr
 
-import org.joda.time.{DateTimeZone, DateTime}
-import org.slf4j.{LoggerFactory, Logger}
+import org.joda.time.{DateTime, DateTimeZone}
+import org.slf4j.{Logger, LoggerFactory}
 
+import scala.collection.immutable.Seq
 import scala.xml._
 
 class SolrDocumentGenerator(fedora: FedoraProvider, pid: String, log: Logger = LoggerFactory.getLogger(getClass)) {
   val DC_NAMESPACE: String = "http://purl.org/dc/elements/1.1/"
   val EAS_NAMESPACE: String = "http://easy.dans.knaw.nl/easy/easymetadata/eas/"
   val RDF_NAMESPACE: String = "http://www.w3.org/1999/02/22-rdf-syntax-ns#"
 
-  val dc = XML.loadString(fedora.getDc(pid))
   val emd = XML.loadString(fedora.getEmd(pid))
   val amd = XML.loadString(fedora.getAmd(pid))
   val prsl = XML.loadString(fedora.getPrsql(pid))
   val relsExt = XML.loadString(fedora.getRelsExt(pid))
 
-  val dcMappings = List("title", "description", "creator", "subject", "publisher",
-    "contributor", "date", "type", "format", "identifier", "source", "language", "relation",
-    "coverage", "rights")
-    .map(s => s"dc_$s" -> (dc \ s).map(_.text))
+  /* dc */
 
-  val dcMappingsSort = List("title", "creator", "publisher", "contributor")
-    .map(s => s"dc_${s}_s" -> (dc \\ s).map(_.text))
+  /* with sort fields */
+
+  def extract(name: String)(f: Node => String): (String, Seq[String]) = {
+    s"dc_$name" -> (emd \ name \ "_").map(f)
+  }
+
+  val dcTitleFromEmdMapping@(dcTitleKey, dcTitleValues) = extract("title")(_.text)
```

Comment:
That is variable binding and pattern matching. See: http://www.artima.com/pins1ed/case-classes-and-pattern-matching.html#lst:variable-binding

Had plans to cover this in a workshop on patter matching, but I can as well refer to the link above.